airbyte-integrations/bases/base-python/base_python/cdk/streams/http.py

```diff
@@ -37,7 +37,8 @@ class HttpStream(Stream, ABC):
     Base abstract class for an Airbyte Stream using the HTTP protocol. Basic building block for users building an Airbyte source for a HTTP API.
     """
 
-    source_defined_cursor = True  # Most HTTP streams use a source defined cursor (i.e: the user can't configure it like on a SQL table)
+    cursor = True  # Most HTTP streams use a source defined cursor (i.e: the user can't configure it like on a SQL table)
+    primary_key = "" # Change this to the field this stream should use as a primary key. Use a list if the key should be formed from multiple fields.
```

Comment:
why set this to an empty string? I think either we make the base property not abstract and return None, or require users to choose a primary key. I'm kind of on the side of making it abstract and not overriding it here because almost all data has a primary key, so simply asking the user about the primary key will result in a much higher rate of connectors declaring primary keys.

Reply:
good point.